tests/jballerina-unit-test/src/test/resources/test-src/types/table/record-type-table-key.bal

```diff
@@ -193,12 +193,52 @@ function testInferTableTypeV2() {
 const int id = 1;
 
 function testVariableNameFieldAsKeyField() {
-    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
     table<record {readonly int id; string name;}> key (id) tb = table [
         {id, name: "Jo"},
         {id: 2, name: "Amy"}
     ];
-    assertEquality(expected, tb.toString());
+    assertEquality(tb[1], <record {readonly int id; string name;}> {"id": 1, "name": "Jo"});
+    assertEquality(tb[2], <record {readonly int id; string name;}> {"id": 2, "name": "Amy"});
+}
+
+function testDefaultValueFieldAsKeyField() {
+    table<record {readonly int id = 1; string name;}> key (id) tb = table [
+        {id, name: "Jo"},
+        {id: 2, name: "Amy"}
+    ];
+    assertEquality(tb[1], <record {readonly int id = 1; string name;}> {"id": 1, "name": "Jo"});
+    assertEquality(tb[2], <record {readonly int id = 1; string name;}> {"id": 2, "name": "Amy"});
```

Comment:
The default isn't required in these inline record typedescs.

Reply:
Updated